clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/security/KubernetesApiClientConfig.groovy

```diff
@@ -22,18 +22,48 @@ import io.kubernetes.client.util.KubeConfig;
 import io.kubernetes.client.util.SSLUtils;
 
 import groovy.util.logging.Slf4j
+import org.yaml.snakeyaml.Yaml
+import org.yaml.snakeyaml.constructor.SafeConstructor
 
 import javax.net.ssl.KeyManager;
 
 @Slf4j
 public class KubernetesApiClientConfig extends Config {
   String kubeconfigFile
+  String context
+  String cluster
+  String user
 
-  public KubernetesApiClientConfig(String kubeconfigFile) {
+  public KubernetesApiClientConfig(String kubeconfigFile, String context, String cluster, String user) {
     this.kubeconfigFile = kubeconfigFile
+    this.context = context
+    this.user = user
   }
 
   public ApiClient getApiCient() throws Exception {
-    return (kubeconfigFile ? fromConfig(kubeconfigFile) : Config.defaultClient())
+    InputStream is = new FileInputStream(kubeconfigFile);
+    Reader input = new InputStreamReader(is);
+    Yaml yaml = new Yaml(new SafeConstructor());
+    Object config = yaml.load(input);
+    Map<String, Object> configMap = (Map<String, Object>)config;
+
+    //Override these values
+    if (this.context) {
+      configMap.put("current-context", this.context)
+    }
+
+    if (this.cluster) {
+      configMap.put("clusters", cluster)
+    }
+
+    if (this.user) {
+      configMap.put("users", user)
+    }
+
+    yaml = new Yaml(new SafeConstructor());
+    String out = yaml.dump(configMap);
+    InputStream kubeconfigIS = new ByteArrayInputStream(out.getBytes());
```

Comment:
What is the goal here? You don't need to recreate a kubeconfig. If a user has specified a context, it's assumed it exists in their kubeconfig, not the one they are creating.

Reply:
If the value is presenting is clouddriver-local.yml.  It should be allow to override.  Is it? or we just need to update if the value does not present in the config file.

Reply:
Yes it should override `current-context`. That doesn't mean we need to create a new kubeconfig object, though.